Mathlib/AlgebraicGeometry/Pullbacks.lean

```diff
@@ -536,6 +536,62 @@ def openCoverOfBase (𝒰 : OpenCover Z) (f : X ⟶ Z) (g : Y ⟶ Z) : OpenCover
       PullbackCone.π_app_right, IsPullback.cone_snd, pullbackSymmetry_hom_comp_fst_assoc]
     rfl
 
+variable (f : X ⟶ Y) (𝒰 : Y.OpenCover) (𝒱 : ∀ i, ((𝒰.pullbackCover f).obj i).OpenCover)
+
+/--
+Given `𝒰 i` covering `Y` and `𝒱 i j` covering `𝒰 i`, this is the open cover
+`𝒱 i j₁ ×_{𝒰 i} 𝒱 i j₂` ranging over all `i`, `j₁`, `j₂`.
+-/
+noncomputable
+def diagonalCover : (pullback.diagonalObj f).OpenCover :=
+(openCoverOfBase 𝒰 f f).bind
+  fun i ↦ openCoverOfLeftRight (𝒱 i) (𝒱 i) (𝒰.pullbackHom _ _) (𝒰.pullbackHom _ _)
+
+/-- The image of `𝒱 i j₁ ×_{𝒰 i} 𝒱 i j₂` in `diagonalCover` with `j₁ = j₂`  -/
+noncomputable
+def diagonalCoverDiagonal : (pullback.diagonalObj f).Opens :=
+⨆ i : Σ i, (𝒱 i).J, ((diagonalCover f 𝒰 𝒱).map ⟨i.1, i.2, i.2⟩).opensRange
```

Comment:
```suggestion
  (openCoverOfBase 𝒰 f f).bind
    fun i ↦ openCoverOfLeftRight (𝒱 i) (𝒱 i) (𝒰.pullbackHom _ _) (𝒰.pullbackHom _ _)

/-- The image of `𝒱 i j₁ ×_{𝒰 i} 𝒱 i j₂` in `diagonalCover` with `j₁ = j₂`  -/
noncomputable
def diagonalCoverDiagonal : (pullback.diagonalObj f).Opens :=
  ⨆ i : Σ i, (𝒱 i).J, ((diagonalCover f 𝒰 𝒱).map ⟨i.1, i.2, i.2⟩).opensRange
```

Reply:
I don't understand the name `diagonalCoverDiagonal`. Maybe `diagonalCoverDiagonalRange`?

Reply:
This is the diagonal-portion of the diagonal cover. Yeah maybe `diagonalCoverDiagonalRange` works better.